tests/e2e/runtime/apis.spec.ts

```diff
@@ -0,0 +1,39 @@
+import * as puppeteer from "puppeteer";
+import { expect } from "chai";
+import { Utils } from "../../utils";
+import { BrowserLaunchOptions } from "../../constants";
+import { Server } from "http";
+import { Apis } from "../../mocks/collection/apis";
+import { Api } from "../../mocks/collection/api";
+import { ApisWidget } from "../maps/apis";
+
+describe("Apis page", async () => {
+    let config;
+    let browser: puppeteer.Browser;
+    let server: Server;
+    
+    before(async () => {
+        config = await Utils.getConfig();
+        browser = await puppeteer.launch(BrowserLaunchOptions);
+    });
+    after(async () => {
+        browser.close();
+        Utils.closeServer(server);
+    });
+
+    it("User can see apis on the page", async () => {
+        var apis = new Apis();
+        apis.addApi(Api.getEchoApi());
+        server = await Utils.createMockServer([apis.getApisListResponse()]);
+
+        const page = await browser.newPage();
+        await page.goto(config.urls.apis);
+
+        const apiWidget = new ApisWidget(page);
+        await apiWidget.apis();
+
+        expect(await page.evaluate(() =>
+            document.querySelector("api-list div.table div.table-body div.table-row")?.parentElement?.childElementCount
+        )).to.equal(apis.apiList.length);
```

Comment:
I would suggest as you have a great opportunity to create a great example of test. In my opinion two elements are missing.

1.  Abstraction to set backend data - Abstraction which would make creating data easy and test very readable.
2.  Object oriented representation of page and page elements - This is for future. As tests will get bigger and more complex parts will be reused in many tests. It will be huge time saver in the future.

Please compare the following snipped with current test code. 

```typescript
if("User can see apis on the page", async () => {
  var backendData = Banked.createApi(TestApis.Create());
  
  const portal: DeveloperPortal = await DeveloperPortal.create(browser);
  const apisPage: ApisPage = await portal.openApisPage();
  const apisWidget: ApisWidget = await apisPage.getApisWidget();
  expect(await apisWidget.getListLength()).to.equal(backendData.apis.length);
  // or just asking ApisPage for most common elements/operations
  expect(await apisPage.getApiListLenght()).to.equal(backendData.apis.length);
});
```

Reply:
Thank you @Mielek. Totally agree with both suggestions, and the example looks great, do you mind if it will be done after this pr gets merged so we could populate by tests + get more feedback on how it is and implement the suggestions all at once as an improvement?

Reply:
I am ok with that, just want to make sure that it is not lost and it will be done. This is a big opportunity to have a great guideline how to properly do such tests.